AutoCollection/CorrelationContextManager.ts

```diff
@@ -35,16 +36,16 @@ export class CorrelationContextManager {
     /**
      *  A helper to generate objects conforming to the CorrelationContext interface
      */
-    public static generateContextObject(parentId?: string, operationName?: string, operationId?: string): CorrelationContext {
-        operationId = operationId || Util.newGuid();
+    public static generateContextObject(operationId: string, parentId?: string, operationName?: string, correlationContextHeader?: string): CorrelationContext {
         parentId = parentId || operationId;
         
         if (this.enabled) {
             return {
                 operation: {
                     name: operationName,
                     id: operationId,
-                    parentId: parentId
+                    parentId: parentId,
+                    correlationContextHeader: correlationContextHeader
```

Comment:
Is that the property bag header? It should fill `CorrelationContext.customProperties`. That was the placeholder object intended for this. The idea is to deserialize it into a dictionary so that its user-editable, and then re-serialize it when outgoing